src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js

```diff
@@ -1,28 +1,60 @@
 // @flow
-import React from 'react';
+import React, { useCallback } from 'react';
 // $FlowFixMe
-import { useSelector, shallowEqual } from 'react-redux';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import {
+    addEnrollmentEventPageActionTypes,
+    navigateToEnrollmentPage,
+} from './enrollmentAddEventPage.actions';
 import { useProgramInfo } from '../../../hooks/useProgramInfo';
 import { EnrollmentAddEventPageComponent } from './EnrollmentAddEventPage.component';
 
 export const EnrollmentAddEventPage = () => {
-    const { programId, stageId } = useSelector(
+    const { programId, stageId, orgUnitId, teiId, enrollmentId } = useSelector(
         ({
             router: {
                 location: { query },
             },
         }) => ({
             programId: query.programId,
             stageId: query.stageId,
+            orgUnitId: query.orgUnitId,
+            teiId: query.teiId,
+            enrollmentId: query.enrollmentId,
         }),
         shallowEqual,
     );
+
+    const dispatch = useDispatch();
+
+    const handleCancel = useCallback(() => {
+        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    const handleSave = useCallback(() => {
+        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    // TODO: Validate query params
     const { program } = useProgramInfo(programId);
     const programStage = [...program.stages.values()].find(item => item.id === stageId);
     if (!programStage) {
         return <span>[program stage placeholder]</span>;
     }
+
+    // TODO: Get data from enrollment collection for the rules engine
```

Comment:
Is this todo related to https://jira.dhis2.org/browse/TECH-635?

Reply:
Yes, it is :+1. Ticket-number added to comment.